WordPress/MediaEditor/MediaEditorHub.swift

```diff
@@ -78,16 +117,134 @@ class MediaEditorHub: UIViewController {
+        imagesCollectionView.scrollToItem(at: IndexPath(row: self.selectedThumbIndex, section: 0), at: .right, animated: false)
+        thumbsToolbar.isHidden = numberOfThumbs > 1 ? false : true
+    }
+
     private func setupForOrientation() {
         let isLandscape = UIDevice.current.orientation.isLandscape
-        portraitConstraints.forEach { $0.isActive = !isLandscape }
-        landscapeConstraints.forEach { $0.isActive = isLandscape }
+        mainStackView.axis = isLandscape ? .horizontal : .vertical
+        mainStackView.semanticContentAttribute = isLandscape ? .forceRightToLeft : .unspecified
         horizontalToolbar.isHidden = isLandscape
         verticalToolbar.isHidden = !isLandscape
+        if let layout = thumbsCollectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.scrollDirection = isLandscape ? .vertical : .horizontal
+        }
+        mainStackView.layoutIfNeeded()
+        imagesCollectionView.scrollToItem(at: IndexPath(row: selectedThumbIndex, section: 0), at: .right, animated: false)
+    }
+
+    private func highlightSelectedThumb(current: Int, before: Int) {
+        let current = thumbsCollectionView.cellForItem(at: IndexPath(row: current, section: 0)) as? MediaEditorThumbCell
+        let before = thumbsCollectionView.cellForItem(at: IndexPath(row: before, section: 0)) as? MediaEditorThumbCell
+        before?.hideBorder()
+        current?.showBorder()
+    }
+
+    private func showOrHideActivityIndicator() {
+        let imageAvailable = availableThumbs[selectedThumbIndex] ?? availableImages[selectedThumbIndex]
+
+        imageAvailable == nil ? showActivityIndicator() : hideActivityIndicator()
     }
 
     static func initialize() -> MediaEditorHub {
         return UIStoryboard(name: "MediaEditorHub", bundle: nil).instantiateViewController(withIdentifier: "hubViewController") as! MediaEditorHub
     }
 
+    private enum Constants {
+        static var thumbCellIdentifier = "thumbCell"
+        static var imageCellIdentifier = "imageCell"
+    }
+}
+
+// MARK: - UICollectionViewDataSource
+
+extension MediaEditorHub: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return numberOfThumbs
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == thumbsCollectionView {
+            return cellForThumbsCollectionView(cellForItemAt: indexPath)
+        }
+
+        return cellForImagesCollectionView(cellForItemAt: indexPath)
+    }
+
+    private func cellForThumbsCollectionView(cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = thumbsCollectionView.dequeueReusableCell(withReuseIdentifier: Constants.thumbCellIdentifier, for: indexPath) as? MediaEditorThumbCell else {
+            return UICollectionViewCell()
```

Comment:
This case happening would likely mean that the identifier provided above is not right, or it doesn't match the one we used for registering the cells.  Either way, this case triggering would signal something's off.

As such I'd suggest to log an error here, before returning.

If you use this method you can log an error with Sentry without that error killing the App or interrupting the user:

https://github.com/Automattic/Automattic-Tracks-iOS/blob/b8e747a37cf5731c77ba7832e3ef54f540740e60/Automattic-Tracks-iOS/CrashLogging.swift#L110

Reply:
If we return a `UICollectionViewCell()` directly (without using `dequeueReusableCell`) the app will crash anyway (and the crash will be reported to Sentry).

I refactored those methods to make it simpler and close to other examples we have in the app. :)